refactor(rust): renaming project cmd ticket-ttl arg to expires-in

seve-martinez opened the pull request:
<!-- Thank you for sending a pull request :heart: -->

## Current behavior
Closes #5673.

The `project` command takes `ticket-ttl` as an argument. 
```
$ project ticket -h
Add members to a project as an authorised enroller

Usage: ockam project ticket [OPTIONS]

Options:
  ...
      --ticket-ttl <SECS>
```

<!-- Please describe the current behavior of the code before the changes in this pull request is applied. -->

## Proposed changes

Rename this argument to `expires-in`.

```
$ ockam project ticket -h
Add members to a project as an authorised enroller

Usage: ockam project ticket [OPTIONS]

Options:
      ...
      --expires-in <SECS>
```

<!-- Please describe the changes proposed in this pull request. -->
<!-- If this pull request resolves an already recorded bug or a feature request, please add a link to that issue. -->

## Checks

<!-- To help us review and merge this pull request quickly, please confirm the following:  -->

- [X] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [X] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [X] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [X] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

<!-- Looking forward to merging your contribution!! -->

seve-martinez committed:
refactor(rust): renaming project cmd ticket-ttl arg to expires-in

mrinalwadhwa commented:
@seve-martinez thank you for sending the pull request. There is a failing [test](https://github.com/build-trust/ockam/actions/runs/5914045555/job/16038965819?pr=5677#step:7:21). The following bats test also needs an update in two places. Could you change that as well please

https://github.com/build-trust/ockam/blob/2513cc701aacc9ccb32c715b096aa7d093abd1bd/implementations/rust/ockam/ockam_command/tests/bats/authority.bats#L101-L110

seve-martinez commented:
> @seve-martinez thank you for sending the pull request. There is a failing [test](https://github.com/build-trust/ockam/actions/runs/5914045555/job/16038965819?pr=5677#step:7:21). The following bats test also needs an update in two places. Could you change that as well please


Thank you for the catch! I've updated the bats as well.

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/project/ticket.rs:
```diff
@@ -50,8 +50,8 @@ pub struct TicketCommand {
-    #[arg(long = "ticket-ttl", value_name = "SECS", conflicts_with = "member")]
-    ticket_ttl: Option<u64>,
+    #[arg(long = "expires-in", value_name = "SECS", conflicts_with = "member")]
+    expires_in: Option<u64>,
```
I'd recommend to switch the `u64` to a `Duration`, and parse it with the `duration_parse` so we can write the units in the parameter (e.g. `1m` or `20s`).

[Here's an example](https://github.com/build-trust/ockam/blob/2513cc701aacc9ccb32c715b096aa7d093abd1bd/implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs#L64) of another command that is doing this.

seve-martinez replied:
What should the  the new arg `value_name` be? `Duration`?

adrianbenavides replied:
Yeah, `Option<Duration>` to be precise.

seve-martinez replied:
Sorry, I meant this line:
https://github.com/build-trust/ockam/blob/1d227c130fe0dc225eac19dd1111d6efc9be77b9/implementations/rust/ockam/ockam_command/src/project/ticket.rs#L54

I assume that `value_name` here should be something other than `SECS` now that we can offer a duration. Perhaps `DURATION` is sufficient?

adrianbenavides replied:
Sorry, I missread the comment yesterday! Exactly, `DURATION` would make most sense.

etorreborre commented:
Great job @seve-martinez! Would you mind adding an example of using this renamed parameter in [the help file](https://github.com/build-trust/ockam/blob/121da2d8ba5e90ae056ee32b4863b52a1d651f65/implementations/rust/ockam/ockam_command/src/project/static/ticket/after_long_help.txt#L4) as well? Thanks!

etorreborre commented:
@seve-martinez you will also need to format the code with `cargo fmt` and rebase to get rid of the error with the `cargo-deny` job.

etorreborre commented:
@seve-martinez I pushed some changes for the failing jobs. While I was trying out the new command parameter I also realized that we have a very limited set of expressions for durations since we only accept minutes, seconds and milliseconds.

However in the case of ticket creation we would like to create longer lived tickets, possibly in the order of days.
Would you mind extending this PR a bit to change the [`Duration` parser](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/util/duration.rs#L11) to also accept `h` for "hour" and `d` for "day"? Thank you 🙏

seve-martinez committed:
refactor(rust): adding hour and day ticket duration

seve-martinez commented:
@etorreborre No problem! I've updated the duration parser function to take in `hour` and `day` as well. I've also updated the `test_reverse_duration`. 

Thanks for the help 🚀

etorreborre reviewed (approved):
This is great, thanks for your contribution!
